src/main/scala/io/github/spark_redshift_community/spark/redshift/RedshiftRelation.scala

```diff
@@ -143,12 +144,18 @@ private[redshift] case class RedshiftRelation(
         val s3URI = Utils.createS3URI(cleanedTempDirUri)
         val s3Client = s3ClientFactory(creds)
         val is = s3Client.getObject(s3URI.getBucket, s3URI.getKey + "manifest").getObjectContent
-        val s3Files = try {
-          val entries = Json.parse(new InputStreamReader(is)).asObject().get("entries").asArray()
-          entries.iterator().asScala.map(_.asObject().get("url").asString()).toSeq
-        } finally {
-          is.close()
-        }
+        val s3Files =
+          try {
+            try {
+              val entries = Json.parse(new InputStreamReader(is))
+                .asObject().get("entries").asArray()
+              entries.iterator().asScala.map(_.asObject().get("url").asString()).toSeq
+            } finally {
+              is.close()
+            }
+          } catch {
+            case _: AmazonS3Exception => Seq()
```

Comment:
@lucagiovagnoli 
I agree with you on hiding exceptions. It is generally bad practice as it obscures the root error from the end developer. However, in this situation we are trying to maintain consistent behavior.
As an end-user, I have designed applications that operate on the assumption that an empty query will result in an empty dataframe being returned and I'd like to avoid refactoring all of my team's code to adjust to the new behavior.
I'd be happy to give users the option to have an error thrown in this situation via some sort of failFast flag.
Finally - you're right that we should let the user know what's going on. I'll add some logging to make it more clear.